src/System.Buffers.ReaderWriter/System/Buffers/Reader/BufferReader_search.cs

```diff
@@ -84,6 +86,102 @@ private bool TryReadToSlow(out ReadOnlySpan<T> span, T delimiter, T delimiterEsc
             return true;
         }
 
+        private bool TryReadToSlow(out ReadOnlySequence<T> sequence, T delimiter, T delimiterEscape, int index, bool advancePastDelimiter)
+        {
+            BufferReader<T> copy = this;
+
+            ReadOnlySpan<T> remaining = UnreadSpan;
+            bool priorEscape = false;
+
+            do
+            {
+                if (index >= 0)
+                {
+                    if (index == 0 && priorEscape)
+                    {
+                        // We were in the escaped state, so skip this delimiter
+                        priorEscape = false;
+                        Advance(index + 1);
+                        goto Continue;
+                    }
+                    else if (index > 0 && remaining[index - 1].Equals(delimiterEscape))
+                    {
+                        // This delimiter might be skipped
+
+                        // Count our escapes
+                        int escapeCount = 1;
+                        int i = index - 2;
+                        for (; i >= 0; i--)
+                        {
+                            if (!remaining[i].Equals(delimiterEscape))
+                                break;
+                        }
+                        if (i < 0 && priorEscape)
+                        {
+                            // Started and ended with escape, increment once more
+                            escapeCount++;
+                        }
+                        escapeCount += index - 2 - i;
```

Comment:
Avoiding incrementing escapeCount for each iteration of the loop above. Instead doing it once at the end.